analytics/src/main/java/com/segment/analytics/SegmentIntegration.java

```diff
@@ -182,22 +183,36 @@ static synchronized SegmentIntegration create(
             Logger logger,
             Crypto crypto,
             ValueMap settings) {
-        PayloadQueue payloadQueue;
+        final PayloadQueue[] payloadQueue = new PayloadQueue[1];
+        Thread execThread =
+                new Thread(
+                        () -> {
+                            try {
+                                File folder =
```

Comment:
This seems like a great candidate to use Futures. Futures allow you to return values from a thread. I would suggest using those here

Reply:
also i just wanted to confirm, but are there are any potential downfalls to creating this file from a random thread? I cant think of any but definitely something to think about

Reply:
When I was making it, I didn't think there were any, but I also didn't look into the potential issues of isolating the process to it's own thread... from what I remember in school, as long as there isn't any other process in the thread it shouldn't deadlock (without mutexes or anything to manage it) so I think its okay?